lib/transaction/trace/index.js

```diff
@@ -366,6 +339,85 @@ Trace.prototype._getRequestUri = function _getRequestUri() {
   return requestUri
 }
 
+Trace.prototype.getChildren = function getChildren(id) {
+  return this.segments.filter((segment) => segment.parentId === id)
+}
+
+Trace.prototype.getCollectedChildren = function getCollectedChildren(id) {
+  return this.segments.filter(
+    (segment) => segment.parentId === id && segment._collect && !segment.ignore
+  )
+}
+
+Trace.prototype.getParent = function getParent(parentId) {
+  return this.segments.filter((segment) => segment.id === parentId)[0]
+}
+
+/**
+ * This is perhaps the most poorly-documented element of transaction traces:
```

Comment:
I believe this is documented in `Transaction-Events-PORTED` and `analytic_event_data (protocol 17)`?

Reply:
this code used to live in `trace/segment` so i didn't change anything about it